dts/arm/infineon/xmc4500.dtsi

```diff
@@ -21,5 +21,5 @@
 };
 
 &flash0 {
-	reg = <0xc000000 DT_SIZE_M(1)>;
+	reg = <0xc000000 DT_SIZE_M(1) 0x8000000 DT_SIZE_M(1)>;
```

Comment:
So the idea here is to use cached address space always? If so, I need to read back the TRM about flushing the cache during abnormal scenarios like reset.

Reply:
> 

That's how it's done in the Infineon linker script. I didn't see in the manual (or example code startup code) about handling these cases. 

There can be some cache coherency problems when writing/erasing cache and then reading from cached address space..  (but we don't have a flash driver yet):

![Screenshot_2022-07-03_10-12-48](https://user-images.githubusercontent.com/17433307/177044052-c3d3fef5-37d7-4eb4-8ea0-b295bf6955fc.png)

Reply:
> So the idea here is to use cached address space always?

The user can also go back to the previous version (using uncached address space) by setting:
reg = <0xc000000 DT_SIZE_M(1) 0xc000000 DT_SIZE_M(1)>;